lib/manager/npm/post-update/npm.ts

```diff
@@ -20,84 +20,67 @@ export async function generateLockFile(
 ): Promise<GenerateLockFileResult> {
   logger.debug(`Spawning npm install to create ${cwd}/${filename}`);
   const { skipInstalls, postUpdateOptions } = config;
-  let lockFile: string = null;
-  let stdout = '';
-  let stderr = '';
-  let cmd = 'npm';
-  let args = '';
+
+  let lockFile = null;
   try {
-    // istanbul ignore if
-    if (config.binarySource === BinarySource.Docker) {
-      logger.debug('Running npm via docker');
-      cmd = `docker run --rm `;
-      // istanbul ignore if
-      if (config.dockerUser) {
-        cmd += `--user=${config.dockerUser} `;
-      }
-      const volumes = [cwd];
-      if (config.cacheDir) {
-        volumes.push(config.cacheDir);
-      }
-      cmd += volumes.map((v) => `-v "${v}":"${v}" `).join('');
-      if (config.dockerMapDotfiles) {
-        const homeDir =
-          process.env.HOME || process.env.HOMEPATH || process.env.USERPROFILE;
-        const homeNpmrc = join(homeDir, '.npmrc');
-        cmd += `-v ${homeNpmrc}:/home/ubuntu/.npmrc `;
-      }
-      const envVars = ['NPM_CONFIG_CACHE', 'npm_config_store'];
-      cmd += envVars.map((e) => `-e ${e} `).join('');
-      cmd += `-w "${cwd}" `;
-      cmd += `renovate/npm npm`;
-    }
-    logger.debug(`Using npm: ${cmd}`);
-    args = `install`;
+    const preCommands = ['npm i -g yarn'];
+    const commands = [];
+    let cmdOptions = '';
     if (
       (postUpdateOptions && postUpdateOptions.includes('npmDedupe')) ||
       skipInstalls === false
     ) {
-      logger.debug('Performing full npm install');
-      args += ' --ignore-scripts --no-audit';
+      logger.debug('Performing node_modules install');
+      cmdOptions += '--ignore-scripts --no-audit';
     } else {
-      args += ' --package-lock-only --no-audit';
+      logger.debug('Updating lock file only');
+      cmdOptions += '--package-lock-only --no-audit';
+    }
+    const execOptions: ExecOptions = {
+      cwd,
+      extraEnv: {
+        NPM_CONFIG_CACHE: env?.NPM_CONFIG_CACHE,
+        npm_config_store: env?.npm_config_store,
```

Comment:
```suggestion
        NPM_CONFIG_CACHE: env.NPM_CONFIG_CACHE,
        npm_config_store: env.npm_config_store,
```

env is not optional?